src/braket/program_sets/program_set.py

```diff
@@ -206,7 +207,7 @@ def __repr__(self):
         )
 
 
-def _zip_circuit_bindings(
+def _zip_circuit_bindings(  # noqa: C901
```

Comment:
What's the warning?

Reply:
https://www.flake8rules.com/rules/C901.html

If we believe the complexity of this function hinders testing, then let's break it up. If we already have 100% coverage on it, let's leave it, and indeed add the noqa comment.